driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -341,6 +342,8 @@ private void write(Connection connection, Connection.ResponseCallback responseCa
                     break;
             }
 
+            // set this flag to true before writing the request
+            requestSent = true;
```

Comment:
This is a bit too coarse-grained IMO. For exceptions thrown directly from `Connection#write(ResponseCallback, boolean)` and also if we detect an error in `Connection#writeHandler`, we can be sure that the request has not been sent. Thinking of a way to handle that.